game/src/Map.cpp

```diff
@@ -0,0 +1,79 @@
+#include <include/Map.hpp>
+
+Map::Map(const uint16_t &id, const uint8_t &currentType, const std::string &mapName, const std::string &bitmapName, const tileArray &tiles, const uint16_t &dimensionX, const uint16_t &dimensionY, const entityMap &entities) {
+    setId(id);
+    setCurrentType(currentType);
+    setMapName(mapName);
+    setBitmapName(bitmapName);
+    setTiles(tiles, dimensionX, dimensionY);
+    setEntities(entities);
+
+}
+
+Map::~Map() {
+    entities_.clear();
+    
+}
+
+void Map::setId(const uint16_t &id) {
+    this->id_ = id;
+}
+void Map::setCurrentType(const uint8_t &currentType) {
+    this->currentType_ = currentType;
+}
+void Map::setMapName(const std::string &mapName) {
+    this->mapName_ = mapName;
+}
+void Map::setBitmapName(const std::string &bitmapName) {
+    this->bitmapName_ = bitmapName;
+}
+void Map::setTiles(const tileArray &tiles, const uint16_t &dimensionX, const uint16_t &dimensionY) {
+    if(dimensionX == 0 || dimensionY == 0) {
+        return;
+    }
+    tiles_.resize(boost::extents[dimensionX][dimensionY]); 
+    this->tiles_ = tiles;
+}
+void Map::setEntities(const entityMap &entities) {
+    this->entities_ = entities;
+}
+void Map::setTile(const utils::Coordinate &coordinate, const Tile &tile) {
+    tiles_[coordinate.x][coordinate.y] = tile;
+}
+
+const uint16_t Map::getId() {
+    return id_;
+}
+const uint8_t Map::getCurrentType() {
+    return currentType_;
+}
+const std::string Map::getMapName() {
+    return mapName_;
+}
+const std::string Map::getBitmapName() {
+    return bitmapName_;
+}
+const tileArray Map::getTiles() {
+    return tiles_;
+}
+const entityMap Map::getEntities() {
+    return entities_;
+}
+
+void Map::swapTiles(Tile &tile1, Tile &tile2) {
+    Tile tiletmp = tile1;
+    tile1 = tile2;
+    tile2 = tiletmp;
+}
+
+bool Map::doesItCollide(const utils::Coordinate &coordinate, const uint8_t &direction) {
+    Tile tile = tiles_[coordinate.x][coordinate.y];
```

Comment:
We need to be careful here, the coordinate should be the target tile and not the tile where the player is standing.

This is the responsibility of the Player/NPC/Trainer that will call this function. We should absolutely write a test

Reply:
Yes
When developing, I was thinking this way: the Trainer has also a Direction member (aka the direction where the sprite is looking to). We can calculate in which tile is it looking. For example:
Trainer in x=2, y=2, direction=UP
-> Trainer is trying to interact/collide with x=2,y=3
-> x=2,y=3 become the input coordinates of "doesItCollide(Interact)"

Since the Tile doesn't know where the Trainer is looking, but the Trainer does, it should be his responsibility.